### DISCUSSION TITLE
How to collect form values inside Drawer?

### DISCUSSION THREAD

**Discussion by joshua-lehmann** [ant-design/ant-design#31192] [Q&A]
### Reproduction link
[![Edit on CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/submit-form-in-drawer-antd4166-forked-k4zud?file=/index.js)

### Steps to reproduce
Click the "New Account" button, enter something in the form and click "submit" button. 

The goal: Create a simple create form inside a drawer with a submit button in the drawer footer. The submit button should then submit the form and collect the values of the form.

### What is expected?
The form should be submitted (onFormFinish() should be called) and values printed in the console.

### What is actually happening?
The form is not submitted, and nothing happens

| Environment | Info |
|---|---|
| antd | 4.16.6 |
| React | 17.0.2 |
| System | Windows 10 |
| Browser | Google Chrome |

---
I used the [example from the official documentation](https://ant.design/components/drawer/#components-drawer-demo-form-in-drawer) as starting point.
It even states that you can use drawers to create a "modal" form. However, it does not show how to then collect the values from the form with a button in the footer!

**Comment by KiritoXF** [comment]
Hi.
The buttons in footer are not in `Form.Item`, they are out of the form so that we could not use `htmlType="submit"` to submit the form. It won't trigger `onFinish` event.

If you want the buttons in footer, you could use `onClick` to bundle some function. (However, I'm not familiar with react class component)

Or, maybe set `footer={null}` and move the code into the form.
